lib/adagrams.rb

```diff
@@ -0,0 +1,109 @@
+# wave1 code
+def draw_letters
+  pool_of_letters = {
+    "A" => 9, "B" => 2, "C" => 2, "D" => 4, "E" => 12, "F" => 2, "G" => 3, "H" => 2, "I" => 9, "J" => 1, "K" => 1, "L" => 4, "M" => 2, "N" => 6, "O" => 8, "P" => 2, "Q" => 1, "R" => 6, "S" => 4, "T" => 6, "U" => 4, "V" => 2, "W" =>2, "X" => 1, "Y" => 2, "Z" => 1
+  }
+
+  players_letters = []
+  alphabet = ("A".."Z").to_a
+  
+  until players_letters.length == 10
+    random_letter = alphabet.sample
+    if pool_of_letters[random_letter] && pool_of_letters[random_letter] > 0
+      players_letters << random_letter
+      pool_of_letters[random_letter] -= 1
+    end
+  end
+  return players_letters
+end
+
+# wave2 code
+def uses_available_letters?(input, letters_in_hand)
```

Comment:
Review this implementation of this method that does not require the creation of new hashes:

```ruby
def uses_available_letters?(input, letters_in_hand)
  input_array = input.split('')
  letters_in_hand_copy = letters_in_hand.shuffle

  input_array.each do |letter_in_word|
    if letters_in_hand_copy.include?(letter_in_word)
      letters_in_hand_copy.delete(letter_in_word)
    else
      return false
    end
  end
  
  return true
end
```